### PR TITLE
[build] PT1.5.0 for python and awscli version

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
- PT1.5.0 for python and awscli version

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -90,8 +91,14 @@ RUN pip install --no-cache-dir "sagemaker-pytorch-inference<2"
...
-RUN conda install -y -c conda-forge pyyaml==5.3.1
-RUN pip install pillow==7.1.0 "awscli<2"
+# install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+RUN conda install -y -c conda-forge "PyYAML>=5.4,<5.5"
+RUN pip install \
+    pillow==7.1.0 \
+    "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2" \
+    ruamel-yaml
```
Comment (saimidu) [inline pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu:101]: Please put `ruamel-yaml` in line 48, and pin it to the current version.

File: pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 2]
```diff
@@ -104,8 +105,15 @@ RUN pip install --no-cache-dir "sagemaker-pytorch-inference<2"
...
-RUN pip install pillow==7.1.0 "awscli<2"
+# install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+RUN conda install -y -c conda-forge "PyYAML>=5.4,<5.5"
+
+RUN pip install \
+    pillow==7.1.0 \
+    "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2" \
+    ruamel-yaml
```
Comment (saimidu) [inline pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu:116]: Please put `ruamel-yaml` in line 59, and pin it to the current version.

File: pytorch/training/docker/1.5.0/py3/Dockerfile.cpu [hunk 3]
```diff
@@ -117,7 +121,10 @@ RUN curl -o /license.txt https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.5.0
 
 # fix safety issues
 RUN pip install psutil==5.6.7 \
-                Pillow==7.1.0
+                Pillow==7.1.0 \
+                # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+                "cryptography>3.2" \
+                ruamel-yaml
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/Dockerfile.cpu:127]: Please put `ruamel-yaml` in line 77, and pin it to the current version.

File: pytorch/training/docker/1.5.0/py3/Dockerfile.cpu [hunk 4]
```diff
@@ -95,11 +97,13 @@ RUN curl -L -o ~/miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-lat
...
  && ln -s /opt/conda/bin/pip /usr/local/bin/pip3 \
  && pip install --no-cache-dir -U \
+    # install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+    "PyYAML>=5.4,<5.5" \
     "awscli<2" \
     fastai==1.0.59 \
     scipy==1.2.2 \
     smdebug==${SMDEBUG_VERSION} \
-    "sagemaker<2" \
+    "sagemaker>=2,<3" \
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/Dockerfile.cpu:106]: Please verify if this is okay.

File: pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # The tag for the base image is: 10.1-cudnn7-devel-ubuntu16.04
-FROM nvidia/cuda@sha256:4979db047661dc0003594fb20d37cce6d6c7e989252f4e3fb0beb39874a078e2
+FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu16.04
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu:2]: Thanks.

File: pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 6]
```diff
@@ -126,11 +128,16 @@ RUN /opt/conda/bin/conda config --set ssl_verify False \
...
+    "sagemaker>=2,<3" \
     sagemaker-experiments==0.* \
     --no-cache-dir fastai==1.0.59 \
+    # install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+    "PyYAML>=5.4,<5.5" \
     "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2" \
     scipy==1.2.2 \
+    ruamel-yaml \
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu:140]: Please put `ruamel-yaml` in line 92, and pin it to the current version.

File: pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 7]
```diff
@@ -126,11 +128,16 @@ RUN /opt/conda/bin/conda config --set ssl_verify False \
 # Uninstall torch and torchvision before installing the custom versions from an S3 bucket
 RUN pip install \
     --no-cache-dir smdebug==${SMDEBUG_VERSION} \
-    "sagemaker<2" \
+    "sagemaker>=2,<3" \
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu:131]: Please verify if this is okay.

File: src/config/build_config.py [hunk 8]
```diff
@@ -5,7 +5,7 @@
 # Do remember to revert it back to False before merging any PR (including NEURON dedicated PR)
 ENABLE_NEURON_MODE = False
 # Frameworks for which you want to disable both builds and tests
-DISABLE_FRAMEWORK_TESTS = []
+DISABLE_FRAMEWORK_TESTS = ["tensorflow","mxnet"]
```
Comment (saimidu) [inline src/config/build_config.py:8]: Reviewer Note: To be reverted when PR is merge-ready.

File: pytorch/buildspec.yml [hunk 9]
```diff
@@ -1,7 +1,9 @@
   account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
   region: &REGION <set-$REGION-in-environment>
   framework: &FRAMEWORK pytorch
-  version: &VERSION 1.7.1
+  version: &VERSION 1.5.0
+  cuda_version: &CUDA_VERSION cu101
+  os_version: &OS_VERSION ubuntu16.04
```
Comment (saimidu) [inline pytorch/buildspec.yml:6]: Reviewer Note: To be reverted when PR is merge-ready.

File: test/dlc_tests/conftest.py [hunk 10]
```diff
@@ -17,8 +17,8 @@
 
 from test import test_utils
 from test.test_utils import (
-    is_benchmark_dev_context, get_framework_and_version_from_tag, get_job_type_from_image, is_tf_version,
-    is_below_tf_version, is_below_mxnet_version, is_below_pytorch_version, 
+    is_benchmark_dev_context, get_framework_and_version_from_tag, get_job_type_from_image, is_tf_version,
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:20]: nit: Remove Space at the end of the line.

File: test/dlc_tests/conftest.py [hunk 11]
```diff
@@ -326,12 +326,10 @@ def tf21_and_above_only():
 def mx18_and_above_only():
     pass
 
-
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:329]: nit: Add this space back.

File: test/dlc_tests/conftest.py [hunk 12]
```diff
@@ -326,12 +326,10 @@ def tf21_and_above_only():
...
 @pytest.fixture(scope="session")
-def pt17_and_above_only():
+def pt16_and_above_only():
     pass
 
-
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:334]: nit: Add this space back.

File: test/dlc_tests/sanity/test_torchvision.py [hunk 13]
```diff
@@ -10,7 +10,7 @@
 
 
 @pytest.mark.model("N/A")
-def test_torchvision_nms_training(pytorch_training):
+def test_torchvision_nms_training(pytorch_training, pt16_and_above_only):
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_torchvision.py:13]: This test applies to all PyTorch versions.

It is meant to test that the custom-torchvision binary has been correctly installed in the DLC.

File: test/dlc_tests/sanity/test_torchvision.py [hunk 14]
```diff
@@ -25,7 +25,7 @@ def test_torchvision_nms_training(pytorch_training):
 
 
 @pytest.mark.model("N/A")
-def test_torchvision_nms_inference(pytorch_inference):
+def test_torchvision_nms_inference(pytorch_inference, pt16_and_above_only):
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_torchvision.py:28]: This test applies to all PyTorch versions.

It is meant to test that the custom-torchvision binary has been correctly installed in the DLC.

File: test/test_utils/__init__.py [hunk 15]
```diff
@@ -96,21 +96,7 @@ def is_below_tf_version(version_upper_bound, image_uri):
...
-    Validate that image_uri has framework version strictly less than version_upper_bound
-
-    :param version_upper_bound: str Framework version that image_uri is required to be below
-    :param image_uri: str ECR Image URI for the image to be validated
-    :return: bool True if image_uri has framework version less than version_upper_bound, else False
-    """
-    image_framework_name, image_framework_version = get_framework_and_version_from_tag(image_uri)
-    required_version_specifier_set = SpecifierSet(f"<{version_upper_bound}")
-    return image_framework_name == "mxnet" and image_framework_version in required_version_specifier_set
-
```
Comment (saimidu) [inline test/test_utils/__init__.py:113]: Nice.

File: pytorch/training/docker/1.5.0/py3/Dockerfile.cpu (hunk 4 above)
Comment (tejaschumbalkar) [reply pytorch/training/docker/1.5.0/py3/Dockerfile.cpu]: reverting back the change, keeping the release changes for security patches only.

File: test/dlc_tests/sanity/test_torchvision.py (hunk 13 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_torchvision.py:13]: nms does not seems to be in torchvision for v1.5.0
we get such error `RuntimeError: No such operator torchvision::nms`

Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_torchvision.py:13]: Pytorch team will be built the `torchvision` binary again to solve this issue. Will revert the test skip.

File: test/dlc_tests/sanity/test_torchvision.py (hunk 14 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_torchvision.py:28]: Pytorch team will be re-build the `torchvision` binary again to solve this issue. Will revert the test skip.

File: test/dlc_tests/sanity/test_torchvision.py (hunk 13 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_torchvision.py]: updated the torchvision binary.

File: test/dlc_tests/sanity/test_torchvision.py (hunk 14 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_torchvision.py]: removed the pt16 condition.

File: test/dlc_tests/sanity/test_pre_release.py [hunk 16]
```diff
@@ -258,14 +258,12 @@ def _run_dependency_check_test(image, ec2_connection, processor):
 
 @pytest.mark.model("N/A")
 @pytest.mark.parametrize("ec2_instance_type", ["c5.4xlarge"], indirect=True)
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_pre_release.py:260]: Temp changes. To be reverted

File: test/dlc_tests/sanity/test_safety_check.py [hunk 17]
```diff
@@ -127,9 +127,6 @@ def _get_latest_package_version(package):
 
 @pytest.mark.model("N/A")
 @pytest.mark.skipif(not is_dlc_cicd_context(), reason="Skipping test because it is not running in dlc cicd infra")
```
Comment (tejaschumbalkar) [inline test/dlc_tests/sanity/test_safety_check.py:129]: Temp changes. To be reverted

File: pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu [hunk 18]
```diff
@@ -71,7 +74,7 @@ RUN pip uninstall -y torch \
  && pip install --no-cache-dir -U https://pytorch-aws.s3-us-west-2.amazonaws.com/pytorch-1.5.0/inference/cpu/torch-1.5.0-cp36-cp36m-manylinux1_x86_64.whl \
  && pip uninstall -y torchvision \
  && pip install --no-deps --no-cache-dir -U \
-    https://torchvision-build.s3.amazonaws.com/1.5.0/cpu/torchvision-0.6.0-cp36-cp36m-manylinux1_x86_64.whl
+    https://torchvision-build.s3-us-west-2.amazonaws.com/1.5.0/cpu/torchvision-0.6.0%2Bcpu-cp36-cp36m-linux_x86_64.whl
```
Comment (tejaschumbalkar) [inline pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu:77]: added new binary to resolve the nms issue in CPU inference image

File: pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu [hunk 19]
```diff
@@ -90,8 +93,13 @@ RUN pip install --no-cache-dir "sagemaker-pytorch-inference<2"
...
 
-RUN conda install -y -c conda-forge pyyaml==5.3.1
-RUN pip install pillow==7.1.0 "awscli<2"
+# install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+RUN conda install -y -c conda-forge "PyYAML>=5.4,<5.5"
+RUN pip install \
+    pillow==7.1.0 \
+    "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2"
```
Comment (saimidu) [inline pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu:102]: Please pin to `>=3.3.2`

File: pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 20]
```diff
@@ -104,8 +107,14 @@ RUN pip install --no-cache-dir "sagemaker-pytorch-inference<2"
...
-RUN conda install -y -c conda-forge pyyaml==5.3.1
-RUN pip install pillow==7.1.0 "awscli<2"
+# install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+RUN conda install -y -c conda-forge "PyYAML>=5.4,<5.5"
+
+RUN pip install \
+    pillow==7.1.0 \
+    "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2"
```
Comment (saimidu) [inline pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu:117]: Please pin to >=3.3.2

File: pytorch/training/docker/1.5.0/py3/Dockerfile.cpu [hunk 21]
```diff
@@ -117,7 +123,9 @@ RUN curl -o /license.txt https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.5.0
 
 # fix safety issues
 RUN pip install psutil==5.6.7 \
-                Pillow==7.1.0
+                Pillow==7.1.0 \
+                # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+                "cryptography>3.2"
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/Dockerfile.cpu:128]: Pin to >=3.3.2

File: pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu [hunk 22]
```diff
@@ -129,7 +133,11 @@ RUN pip install \
     "sagemaker<2" \
     sagemaker-experiments==0.* \
     --no-cache-dir fastai==1.0.59 \
+    # install PyYAML>=5.4,<5.5 to avoid conflict with latest awscli
+    "PyYAML>=5.4,<5.5" \
     "awscli<2" \
+    # cryptography required by pyopenssl and paramiko, <=3.2 has safety vulnerabilities
+    "cryptography>3.2" \
```
Comment (saimidu) [inline pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu:140]: Please pin to >=3.3.2

File: test/dlc_tests/sanity/test_pre_release.py [hunk 23]
```diff
@@ -368,6 +366,11 @@ def test_sm_pysdk_2(training):
     :param training: training ECR image URI
     """
 
+    _, image_framework_version = get_framework_and_version_from_tag(training)
+    
+    if Version(image_framework_version) == Version("1.5.0"):
+        pytest.skip("sagemaker version < 2.0 is installled for PT 1.5.0 images")
```
Comment (saimidu) [inline test/dlc_tests/sanity/test_pre_release.py:372]: nit: This is true for 1.5.1 as well.

**Review by saimidu — COMMENTED** [review]
Can approve.

File: test/dlc_tests/sanity/test_pre_release.py (hunk 23 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/sanity/test_pre_release.py:372]: We install `sagemaker>=2,<3` on PT v1.5.1

File: pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu (hunk 19 above)
Comment (tejaschumbalkar) [reply pytorch/inference/docker/1.5.0/py3/Dockerfile.cpu]: done

File: pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu (hunk 20 above)
Comment (tejaschumbalkar) [reply pytorch/inference/docker/1.5.0/py3/cu101/Dockerfile.gpu]: done

File: pytorch/training/docker/1.5.0/py3/Dockerfile.cpu (hunk 21 above)
Comment (tejaschumbalkar) [reply pytorch/training/docker/1.5.0/py3/Dockerfile.cpu]: done

File: pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu (hunk 22 above)
Comment (tejaschumbalkar) [reply pytorch/training/docker/1.5.0/py3/cu101/Dockerfile.gpu]: done

**Review by saimidu — COMMENTED** [review]
APPROVED. Please revert temp changes.

**Review by saimidu — APPROVED** [review]

**Pull request merged**